include/z64map_mark.h

```diff
@@ -0,0 +1,27 @@
+#ifndef _Z64MAP_MARK_H_
+#define _Z64MAP_MARK_H_
+
+#include "ultra64.h"
+
+#define MAP_MARK_ICON_NONE -1
+#define MAP_MARK_ICON_CHEST 0
+#define MAP_MARK_ICON_BOSS 1
+
+typedef struct {
+    /* 0x00 */ s8 chestFlag; // chest icon is only displayed if this flag is not set for the current room
+    /* 0x01 */ u8 x, y; // coordinates to place the icon (top-left corner), relative to the minimap texture
+} MapMarkPoint; // size = 0x3
+
+typedef struct {
+    /* 0x00 */ s8 markType; // 0 for the chest icon, 1 for the boss skull icon, -1 for none
+    /* 0x01 */ u8 count; // number of icons to display
+    /* 0x02 */ MapMarkPoint points[12];
+} MapMarkIconData; // size = 0x26
+
+typedef struct {
+    MapMarkIconData icons[3];
+} MapMarkData; // size = 0x72
```

Comment:
Is there a reason to create a wrapping struct for this? I feel like it adds additional braces and makes things a bit confusing in code for no real gain, but I might be missing something?

Reply:
mainly to eliminate this pattern: ```mapMarkData = &sLoadedMarkDataTable[dungeon][interfaceCtx->mapRoomNum][0]```

Reply:
I'm not sure I see the issue with this pattern, or how using a wrapping struct helps

Reply:
same, can you give an example